src/index.ts

```diff
@@ -37,15 +52,19 @@ export class CSSStyleObserver {
    *
    * @param observedVariables list of CSS variables to observe
    * @param callback callback that will be invoked every time any of listed CSS variables change
+   * @param options configuration options
    */
   constructor(
     observedVariables: string[],
-    callback: CSSStyleObserverCallback
+    callback: CSSStyleObserverCallback,
+    options: CSSStyleObserverOptions = {}
   ) {
     this._observedVariables = observedVariables;
     this._callback = callback;
     this._targetElement = null;
-  }
+    this._callbackMode = options.callbackMode && Object.values(CallbackMode).includes(options.callbackMode) 
+    ? options.callbackMode 
+    : CallbackMode.INDIVIDUAL;  }
```

Comment:
I’m no TypeScript expert, but is this check needed here? I would assume TS to validate the `options.callbackMode` all by itself. If that is the case, then most likely the code can be simplified to this:

```ts
this._callbackMode = options.callbackMode ?? CallbackMode.INDIVIDUAL;
```

Reply:
I’ve simplified the assignment to `this._callbackMode = options.callbackMode ?? CallbackMode.INDIVIDUAL` to rely on TypeScript’s type safety. However, I added a fallback in the _handleUpdate method to default to CallbackMode.INDIVIDUAL if an invalid mode is detected at runtime

Reply:
this assignment can even be simplified to` this._callbackMode = options.callbackMode` as the fallback is handled already in `_handleUpdate` function